# 03 — Modelagem

**Objetivo:** Treinar e comparar modelos de classificação (ex.: Logistic Regression,
Random Forest, SVM), realizar cross-validation e selecionar o melhor modelo.

In [1]:
import sys
sys.path.append('..')

import pandas as pd

from src.data_loader import load_breast_cancer_data
from src.preprocessing import split_and_scale
from src.models import train_model

df = load_breast_cancer_data()
X_train, X_test, y_train, y_test, scaler = split_and_scale(df)

print(f"Treino: {X_train.shape}, Teste: {X_test.shape}")

Treino: (455, 30), Teste: (114, 30)


In [2]:
modelos_nomes = ["logistic_regression", "random_forest", "knn"]

modelos_treinados = {}
for nome in modelos_nomes:
    modelo = train_model(nome, X_train, y_train)
    modelos_treinados[nome] = modelo
    print(f"✓ {nome} treinado")

✓ logistic_regression treinado
✓ random_forest treinado
✓ knn treinado


In [3]:
import joblib
from pathlib import Path

models_dir = Path("../outputs/models")
models_dir.mkdir(parents=True, exist_ok=True)

for nome, modelo in modelos_treinados.items():
    joblib.dump(modelo, models_dir / f"{nome}.pkl")

joblib.dump(scaler, models_dir / "scaler.pkl")

print("Modelos e scaler salvos em outputs/models/")

Modelos e scaler salvos em outputs/models/


## Conclusões da Modelagem

- Três modelos treinados com sucesso sobre o conjunto de treino padronizado
  (455 amostras): Regressão Logística, Random Forest e KNN (k=5).
- Modelos e scaler persistidos em `outputs/models/` para reutilização nas
  etapas de avaliação e interpretabilidade, sem necessidade de retreinamento.

